content/developer/howtos/frontend_owl_components.rst

```diff
@@ -0,0 +1,187 @@
+
+To keep things simple, let's start with a very straightforward component that just renders
+"Hello, World!". This will let us know at a glance if our setup is working.
+
+First, create the template in :file:`/your_module/static/src/portal_component/your_component.xml`.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="UTF-8"?>
+    <templates xml:space="preserve">
+        <t t-name="your_module.YourComponent">
+            Hello, World!
+        </t>
+    </templates>
+
+Then create the JavaScript file for that component in :file:`/your_module/static/src/portal_component/your_component.js`,
+and add it to the `public_components` registry:
+
+.. code-block:: js
+
+    /** @odoo-module */
+    import { Component } from "@odoo/owl";
+    import { registry } from "@web/core/registry"
+
+    export class YourComponent extends Component {
+        static template = "your_module.YourComponent";
+        static props = {};
+    }
+
+    registry.category("public_components").add("your_module.YourComponent", YourComponent);
+
+
+.. seealso::
+   :ref:`Owl components reference<frontend/components>`.
+
+
+2. Adding your component to the `web.assets_frontend` bundle
+============================================================
+
+The `web.assets_frontend` bundle is the assets bundle that is used by the portal and
+website, you'll want to add your component's code to that bundle so that the public
+components service can find your component and mount it. In your module's manifest,
+in the assets section, add an entry for `web.assets_frontend` and add your component's
+files:
+
+.. code-block:: py
+
+    {
+        # ...
+        'assets': {
+            'web.assets_frontend': [
+                'your_module/static/src/portal_component/**/*',
+            ],
+        }
+    }
+
+.. seealso::
+    :ref:`Module manifest reference<reference/module/manifest>`.
+
+3. Adding an `<owl-component>` tag to a page
+============================================
+
+Now we need add an `<owl-component>` tag that will serve as the target for the component
+to be mounted. For the sake of this example, we'll add it directly to the portal's
+home page with an xpath in :file:`/your_module/views/templates.xml`.
+
+.. code-block:: xml
+
+    <?xml version="1.0" encoding="utf-8"?>
+    <odoo>
+        <template id="your_module.portal_my_home" inherit_id="portal.portal_my_home">
+            <xpath expr="//*[hasclass('o_portal_my_home')]" position="before">
+                <owl-component name="your_module.YourComponent" />
+            </xpath>
+        </template>
+    </odoo>
+
+Don't forget to add this file to the data section of your assets bundle:
+
+.. code-block:: py
+
+    {
+        # ...
+        'data': {
+            'views/templates.xml',
+        }
+    }
+
+And that's it! If you open the home page of the portal you should see the message
+"Hello, World!" at the top of the page.
+
+Problems with using Owl components in the portal and website
```

Comment:
There is no need to repeat the page's title. The title could be shorter, something like
```suggestion
Points of caution
```